### PR TITLE
AVFoundationPlayback unification

### PR DESCRIPTION
**Pull request by leandroalonso**
# Goal

To merge the `AVFoundationPlayback` of `iOS` and `tvOS` in a single file.

# How to test

1. Run Clappr (tvOS and iOS)
2. Check the events output. It should start with `ready`, and then some `stalled` and `played` should be displayed

# Additional info

We've fixed 2 issues along with this PR:

1. `Paused` was being triggered after a `didComplete` event. This doesn't happen anymore;
2. In Clappr for `tvOS` the app was not able to listen to all events because the play is automatically triggered. Because of that, the app missed the early events (like `ready`, for example). We fixed that by triggering a play in another thread.

### CODE REVIEW

File: Sources/Clappr/Classes/Base/Playback.swift [hunk 1]
```diff
@@ -96,7 +96,9 @@ open class Playback: UIBaseObject, Plugin {
         }
 
         #if os(tvOS)
-        play()
+        DispatchQueue.main.asyncAfter(deadline: .now()) {
```
Comment (nabilsafatli) [inline Sources/Clappr/Classes/Base/Playback.swift]: Hey,
We are not using `[weak self]` here, maybe this cause a memory leak problem if someone tries to play and dismiss the player right away.

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 2]
```diff
@@ -233,22 +242,47 @@ open class AVFoundationPlayback: Playback {
     fileprivate func setupPlayer() {
         if let asset = self.asset {
             let item: AVPlayerItem = AVPlayerItem(asset: asset)
+            
+            #if os(iOS)
             player = AVPlayer(playerItem: item)
```
Comment (nabilsafatli) [inline Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: Guys, I don't think that we need this `kLoop` anymore.
I suggest use the same structure for tvOS and iOS, `AVQueuePlayer` or `AVPlayer`.
Let's go with `AVPlayer` for both? What do you think?

**Comment by dezouzart** [comment]
@leandroalonso , sometimes at the video's end the playback enter in a "infinity loading" state. Try to seek to the video's end right after video starts.

**Comment by leandroalonso** [comment]
@dezouzart As we've discussed this issue is not caused by our modifications and is currently happening in`dev`.

I believe this is happening because we use a `.mp4` in the example, I wasn't able to reproduce the same issue with `HLS`.

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift (hunk 2 above)
Comment (leandroalonso) [reply Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: Removed.

File: Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift [hunk 3]
```diff
@@ -20,6 +20,15 @@ open class AVFoundationPlayback: Playback {
     private(set) var seekToTimeWhenReadyToPlay: TimeInterval?
 
     @objc internal dynamic var player: AVPlayer?
+    
+    #if os(tvOS)
+    lazy var nowPlayingService: AVFoundationNowPlayingService = {
+        return AVFoundationNowPlayingService()
+    }()
+    
+    fileprivate var playerLooper: AVPlayerLooper?
```
Comment (nabilsafatli) [inline Sources/Clappr/Classes/Plugin/Playback/AVFoundationPlayback.swift]: This was forgotten here.

**Pull request merged**